core/modules/widgets/widget.js

```diff
@@ -557,6 +557,29 @@ Widget.prototype.allowActionPropagation = function() {
 	return true;
 };
 
+Widget.prototype.generateTransclusionFootprint = function() {
+	var parentTransclusionWidget = this.findParentTransclusionWidget(),
+	    node = this,
+	    footprint = node.parentWidget.children.indexOf(node);
+	while(node = node.parentWidget) {
+		if(node === parentTransclusionWidget) {
+			break;
+		} else if(node.parentWidget && node.parentWidget.children) {
+			footprint = footprint + "-" + node.parentWidget.children.indexOf(node);
```

Comment:
It looks like you're using the index of the widget amongst it's siblings as an additional qualifier. Doesn't that mean that the IDs will change if, say, we have a sequence of 10 input elements and then insert a new one at the top?

Reply:
@Jermolene, I think this works ok